src/main/java/com/example/controllers/DashboardController.java

```diff
@@ -0,0 +1,206 @@
+		model.addAttribute("account", account);
+		model.addAttribute("accountMalicious", account);
+
+		return "userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/avatar", method = RequestMethod.GET)
+	public void getImage(final HttpServletResponse response, @RequestParam(value = "image") final String image) {
+
+		File file = storageFacade.exists(image) ? storageFacade.load(image) : storageFacade.load("avatar.png");
+		try {
+			writeResponse(new FileInputStream(file), response.getOutputStream());
+		}
+		catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@RequestMapping(value = "/userDetail/avatar/update", method = RequestMethod.POST)
+	public String updateAvatar(@RequestParam("imageFile") final MultipartFile imageFile, final Principal principal,
+			final RedirectAttributes redirectAttributes) {
+		if (!imageFile.isEmpty()) {
+			try {
+				storageFacade.save(imageFile.getInputStream(), principal.getName() + ".png"); 
+			}
+			catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+
+		redirectAttributes.addAttribute("username", principal.getName());
+
+		return "redirect:/dashboard/userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/certificate", method = RequestMethod.POST)
+	public void getCertificate(final HttpServletResponse response, final Account account) {
+
+		Account bdAccount = accountDao.findUsersByUsername(account.getUsername()).get(0);
+		File tmpFile = null;
+
+		try {
+			// Serialize object
+			tmpFile = File.createTempFile("serial", ".ser");
+			FileOutputStream fos = new FileOutputStream(tmpFile);
+			ObjectOutputStream oos = new ObjectOutputStream(fos);
+			oos.writeObject(new FileUntrustedValid(account.getName()));
+			oos.close();
+			fos.close();
+
+			// Write into response
+			response.setHeader("Content-Disposition", "attachment; filename=\"Certificate_" + bdAccount.getName() + "_" + ".jks\"");
+			int length = writeResponse(new FileInputStream(tmpFile), response.getOutputStream());
+			response.setContentType("application/octet-stream");
+			response.setHeader("Content-Length", length + "");
+
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@RequestMapping(value = "/userDetail/newcertificate", method = RequestMethod.POST)
+	@ResponseBody
+	public String processSimple(@RequestParam(value = "file", required = false) final MultipartFile file, final Model model)
+			throws IOException, ClassNotFoundException {
+
+		ObjectInputStream ois = new ObjectInputStream(file.getInputStream());
+		ois.readObject();
+		ois.close();
+
+		return "<p>File '" + file.getOriginalFilename() + "' uploaded successfully</p>";
+	}
+
+	@RequestMapping(value = "/userDetail/maliciouscertificate", method = RequestMethod.POST)
+	public void getMaliciousCertificate(final HttpServletResponse response, final Account account) {
+
+		Account bdAccount = accountDao.findUsersByUsername(account.getUsername()).get(0);
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 690eb11b87a1b444de55aa444b269b77 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - SQL Injection
_High CWE-89_
Untrusted user-supplied data is inserted into a SQL statement without adequate validation, escaping, or filtering.

A user can change the intent of the SQL query, which may inappropriately disclose or corrupt data within the database.

### How to fix
Rewrite all SQL queries constructed through dynamic concatenation to use an injection-safe query mechanism such as prepared statements with parameterized queries.

Most modern programming languages provide a feature called "parameterized queries" that allow user-supplied data to be inserted safely as values in dynamic SQL queries. Rather than construct the dynamic SQL query by concatenating user-supplied data to static SQL query string fragments, data values are identified in the query by parameter markers or variables. Dynamic data is then passed through a mechanism provided by SQL that prevents the supplied data from changing the meaning of the query.

Note: the exact syntax and use of prepared statements with parameterized queries vary from language to language.